src/test/java/dev/t3d/happraisal/service/InterviewServiceTest.java

```diff
@@ -0,0 +1,99 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.BDDMockito;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class InterviewServiceTest {
+
+  @Mock private InterviewRepository interviewRepository;
+
+  @InjectMocks private InterviewService interviewService;
+
+  @Test
+  void should_create_interview() {
+    // given
+    final Interview expectedInterviewToSave = getInterview();
+
+    // when
+    interviewService.create(expectedInterviewToSave);
+
+    // then
+    var interviewArgumentCaptor = ArgumentCaptor.forClass(Interview.class);
+    BDDMockito.verify(interviewRepository).save(interviewArgumentCaptor.capture());
+    var actualInterviewToSave = interviewArgumentCaptor.getValue();
+    BDDAssertions.then(actualInterviewToSave)
+        .as("Check is an interview has been save")
+        .isEqualTo(expectedInterviewToSave);
+  }
+
+  @Test
+  void should_get_interview_by_id() {
+    // given
+    final Interview expectedInterview = getInterview();
+    var interviewId = UUID.fromString("149e7f49-e66e-4cfc-8485-37993311d9e2");
+    BDDMockito.given(interviewRepository.getById(interviewId)).willReturn(expectedInterview);
+
+    // when
+    final Interview actualInterview = interviewService.getById(interviewId);
+
+    // then
+    BDDAssertions.then(actualInterview)
+        .as("Check is an interview has been return by its id")
+        .isEqualTo(expectedInterview);
+  }
+
+  @Test
+  void should_find_all_interviews() {
+    // given
+    Interview interview = getInterview();
+    BDDMockito.given(interviewRepository.findAll()).willReturn(List.of(interview));
+
+    // when
+    final List<Interview> actualInterviews = interviewService.findAll();
+
+    // when
+    verify(interviewRepository).findAll();
+    BDDAssertions.then(actualInterviews).as("Check the returned interview list size").hasSize(1);
+  }
+
+  private Interview getInterview() {
+    return new Interview(
+        null,
+        LocalDateTime.of(2020, 1, 1, 0, 0, 0, 0).toInstant(ZoneOffset.UTC),
+        new Person(
+            UUID.fromString("e04ac226-ac21-46af-8fd6-c5e808e4c762"), "lastName", "firstName", true),
+        new Person(
+            UUID.fromString("e04ac226-ac21-46af-8fd6-c5e808e4c762"),
+            "lastName",
+            "firstName",
+            false),
+        new Form(
+            UUID.fromString("43793644-4323-407a-9418-279500b42f56"),
+            Set.of(
+                new QuestionAnswer(
+                    UUID.fromString("b78cea0e-8622-42da-86c1-c0f32721c7b1"),
+                    "question",
+                    "answer",
+                    null))));
```

Comment:
For this test in its current form, this `null` has no effects.
In this moment I don't think that the `form` field in `QuestionAnswer` should always be not `null` but for the sake of consistency of the data model, I would modify this creation code instanciating a new `Form` then putting it in the QuestionAnswer constructor instead of this null.

Reply:
Done

Reply:
Done